packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1973,8 +2050,24 @@ export class ContainerRuntime
 				throw new Error("chunkedOp not expected here");
 			case ContainerMessageType.Rejoin:
 				throw new Error("rejoin not expected here");
-			default:
-				unreachableCase(type, `Unknown ContainerMessageType: ${type}`);
+			//* Replace asserts with throw DPE
+			default: {
+				// This should be extremely rare for stashed ops.
+				// It would require a newer runtime stashing ops and then an older one applying them,
+				// e.g. if an app rolled back its container version
+				const compatBehavior = compatDetails.behavior;
+				switch (compatBehavior) {
+					case "FailToProcess":
+						unreachableCase(type, `Unknown ContainerMessageType [${type}]`);
+					case "Ignore":
+						return;
+					default:
+						unreachableCase(
```

Comment:
Is this differentiation valuable or should we just check for `Ignore`?

Reply:
Refactored it, take a look